pkg/snapshot/snapshotprovisioner.go

```diff
@@ -74,6 +84,35 @@ func (p *snapshotProvisioner) snapshotRestore(
 	return pvSrc, labels, err
 }
 
+func (p *snapshotProvisioner) isSnapshotAllowed(
+	snapshot crdv1.VolumeSnapshot,
+	namespace string,
+) bool {
+	allowedNamespaces, ok := snapshot.Metadata.Annotations[storkSnapshotRestoreNamespacesAnnotation]
+	if !ok {
+		return false
+	}
+
+	csvReader := csv.NewReader(strings.NewReader(allowedNamespaces))
+	namespaces, err := csvReader.ReadAll()
+	if err != nil {
+		log.Errorf("Error parsing allowed namespaces: %v", allowedNamespaces)
+		return false
+	}
+
+	for _, namespaceRegEx := range namespaces[0] {
+		regex, err := regexp.Compile(strings.TrimSpace(namespaceRegEx))
+		if err != nil {
+			log.Errorf("Invalid regex for allowed namespaces: %v", namespaceRegEx)
+			return false
+		}
+		if regex.MatchString(namespace) {
```

Comment:
if I have namespaces `dev1` and `dev11` and if the allowed namespaces is just `dev1`, this will match `dev11` too.

Reply:
Added start and end delimiters around each regex which should catch those cases.